### PR TITLE
[Registration] Call update_mailchimp on user update

### PR DESCRIPTION
**Pull request by yonsaber**
There is currently no way for the user to re-sign up for marketing emails, this re-adds that feature in both the normal profile page, as well as the admin page for users.

I did modify some styles and bits but those changes are minimal and more for spacing than anything

### CODE REVIEW

File: app/views/devise/registrations/edit.html.erb [hunk 1]
```diff
@@ -3,40 +3,43 @@
...
+  </div>
+
+  <div class="form-group mb-3">
     <%= f.submit t('.update'), class: 'btn btn-primary' %>
   </div>
 <% end %>
 
-<p><%= link_to t('.cancel_my_account'), registration_path(resource_name), data: { confirm: t('.are_you_sure') }, method: :delete %>.</p>
-
-<%= link_to t('.back'), :back %>
```
Comment (james) [inline app/views/devise/registrations/edit.html.erb:42]: Is there a reason to remove this back link?

**Comment by james** [comment]
This is good, but I'd like to see some tests for it before I merge it in. I can write them in a bit if needs be, but also open to contribution!

File: app/views/devise/registrations/edit.html.erb (hunk 1 above)
Comment (yonsaber) [reply app/views/devise/registrations/edit.html.erb:42]: I can't remember if there was a reason for removing it, happy to restore

**Comment by yonsaber** [comment]
@james happy to write the tests shortly, crashed out earlier as I've just gotten back home today!

**Comment by yonsaber** [comment]
@james updated now with an additional test and updated the existing test as well, could do some more if needed?

File: app/views/devise/registrations/edit.html.erb (hunk 1 above)
Comment (yonsaber) [reply app/views/devise/registrations/edit.html.erb:42]: Restored!

**Comment by james** [comment]
Looks great! Will deploy when I get to my other computer.

**Pull request merged**